xml/api/crm_attribute-2.18.rng

```diff
@@ -20,11 +22,10 @@
             <optional>
                 <attribute name="id"> <text /> </attribute>
             </optional>
+            <attribute name="name"> <text /> </attribute>
+            <attribute name="value"> <text /> </attribute>
             <optional>
-                <attribute name="name"> <text /> </attribute>
-            </optional>
-            <optional>
-                <attribute name="value"> <text /> </attribute>
+                <attribute name="host"> <text /> </attribute>
```

Comment:
Technically making an optional attribute required is backward incompatible, since output that was previously valid could now become invalid. But if no earlier code generated the output without these values before, then it would be fine. Are you sure that's the case?

Reply:
I am pretty sure this is the case.  This message was only ever generated by crm_attribute before.  It has special code to output "(null)" for null, so that should always be set.  And then for name, the only path that called the message also calls cib__read_node_attr first and that function fails with EINVAL if name is null.

...although, crm_attribute doesn't actually do anything in the case where cib__read_node_attr returns EINVAL.  So it is possible to hit the output message with an empty name, which would cause the name to be omitted.  However, that seems like a bug.

And then on the other hand, this was only added at the beginning of March.  Before that, there was no XML for attributes.